extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/WindowOperatorQueryKit.java

```diff
@@ -65,13 +69,23 @@ public QueryDefinition makeQueryDefinition(
       int minStageNumber
   )
   {
-    // need to validate query first
-    // populate the group of operators to be processed as each stage
-    // the size of the operators is the number of serialized stages
-    // later we should also check if these can be parallelized
-    // check there is an empty over clause or not
+    // Need to validate query first.
+    // Populate the group of operators to be processed at each stage.
+    // The size of the operators is the number of serialized stages.
+    // Later we should also check if these can be parallelized.
+    // Check if there is an empty OVER() clause or not.
     List<List<OperatorFactory>> operatorList = new ArrayList<>();
-    boolean isEmptyOverFound = ifEmptyOverPresentInWindowOperstors(originalQuery, operatorList);
+    RowSignature rowSignature = originalQuery.getRowSignature();
+    log.info("Row signature received for query is [%s].", rowSignature);
+
+    boolean isEmptyOverPresent = originalQuery.getOperators()
```

Comment:
A window without a partition key but having an order by key `window (ORDER BY expr)` would still give `isEmptyOverPresent` true, in which case the logic of moving everything to a single partition holds correct. So probably can just change the variable name and the related comments.

Reply:
@sreemanamala The problem is that this logic checks if any empty over() is present - doesn't matter if there are non-empty over() as well in the query.
But this is again an existing logic that I didn't want to touch in this PR, as it'll increase the scope of this PR a lot. Will get to this in a future PR.

Reply:
Discussed this offline with Sree, will be taking this up in the next set of PRs. This doesn't cause any regression, it's an existing logic flow.